### ISSUE TITLE
Error in Vitest with Astro (`No "exports" main defined`)

### ISSUE THREAD

**Issue by risu729** [webpro/knip#328]
## What happens
When I run `npm knip` with `vitest.config.js` with [Astro integration](https://docs.astro.build/en/guides/testing/#vitest), I get an error `No "exports" main defined in repo\node_modules\tsconfck\package.json`.
I tried to look into the problem but could not solve it. The cause is the `jiti` of loading Astro's dependency [`tsconfck`](https://www.npmjs.com/package/tsconfck).
I appreciate it if you could fix this issue, but please ignore it if it is difficult to resolve.

```js
// vitest.config.js
import { getViteConfig } from "astro/config";
export default getViteConfig({});
```

## Versions
Node.js v21.1.0
npm 10.2.0
knip: 2.38.6
astro: 3.4.2
vitest: 0.34.6
\* All are the latest

## Reproduction steps
The minimum reproduction repo is [here](https://github.com/risu729/astro-vitest-knip-test). Just run `npm i` and `npx knip`.
I checked the reproduction both in Windows and Linux.

## Error log
```
[*] Unresolved configuration (from CLI arguments)
{
  cwd: 'C:/Users/takua/GitHub/astro-vitest-knip-test',
  tsConfigFile: undefined,
  gitignore: true,
  isProduction: false,
  isStrict: false,
  isIgnoreInternal: false,
  isShowProgress: false,
  isIncludeEntryExports: false
}
[*] Included workspaces
[ 'astro-vitest-knip-test' ]
[*] Included workspace configs
[
  {
    pkgName: 'astro-vitest-knip-test',
    name: '.',
    config: {
      entry: [
        '{index,cli,main}.{js,mjs,cjs,jsx,ts,tsx,mts,cts}!',
        'src/{index,cli,main}.{js,mjs,cjs,jsx,ts,tsx,mts,cts}!'
      ],
      project: [ '**/*.{js,mjs,cjs,jsx,ts,tsx,mts,cts}!' ],
      paths: {},
      ignore: [],
      ignoreBinaries: [],
      ignoreDependencies: []
    },
    ancestors: []
  }
]
[.] Enabled plugins (.)
[ 'Astro', 'Vitest' ]
[.] Definition paths (0)
[]
[.] Entry paths in package.json (0)
[]
[.] Astro config file paths (0)
[]
[.] Astro dependencies (4)
[
  'entry:astro.config.{js,cjs,mjs,ts}',
  'entry:src/content/config.ts',
  'production:src/content/**/*.mdx',
  'production:src/pages/**/*.{astro,mdx,js,ts}'
]
[.] Vitest config file paths (1)
[ 'C:/Users/takua/GitHub/astro-vitest-knip-test/vitest.config.ts' ]
node:internal/modules/esm/resolve:294
  return new ERR_PACKAGE_PATH_NOT_EXPORTED(
         ^

Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in C:\Users\takua\GitHub\astro-vitest-knip-test\node_modules\tsconfck\package.json
    at exportsNotFound (node:internal/modules/esm/resolve:294:10)
    at packageExportsResolve (node:internal/modules/esm/resolve:584:13)
    at resolveExports (node:internal/modules/cjs/loader:591:36)
    at Module._findPath (node:internal/modules/cjs/loader:668:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1130:27)
    at Function.resolve (node:internal/modules/helpers:187:19)
    at _resolve (C:\Users\takua\GitHub\astro-vitest-knip-test\node_modules\jiti\dist\jiti.js:1:251148)
    at jiti (C:\Users\takua\GitHub\astro-vitest-knip-test\node_modules\jiti\dist\jiti.js:1:253746)
    at C:\Users\takua\GitHub\astro-vitest-knip-test\node_modules\astro\dist\core\config\tsconfig.js:2:17
    at evalModule (C:\Users\takua\GitHub\astro-vitest-knip-test\node_modules\jiti\dist\jiti.js:1:256443) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
```

## Workaround
Add `vitest: false` to `knip.ts`

**Comment by webpro** [comment]
Seeing the same issue in an Astro project when importing `@mdx-js/mdx`:

```
***/knip/dist/util/loader.js:31
        throw new LoaderError(`Error loading ${filePath}`, { cause: error });
              ^

LoaderError: Error loading ***/knip.ts
    at load (***/knip/dist/util/loader.js:31:15)
    at ConfigurationChief.loadResolvedConfigurationFile (***/knip/dist/ConfigurationChief.js:102:35)
    ... 3 lines matching cause stack trace ...
    at async ***/knip/dist/cli.js:73:1 {
  [cause]: Error: No "exports" main defined in ***/node_modules/estree-walker/package.json
```

**Comment by webpro** [comment]
Might be related to or duplicate of #290.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.41.2. See [Release 2.41.2](https://github.com/webpro/knip/releases/tag/2.41.2) for release notes._

**Comment by risu729** [comment]
I confirmed that the issue is resolved. Thank you so much!!!
